### PR TITLE
docs(tls): removes outdated and replicated content related to TLS

### PR DESCRIPTION
**Pull request by PaulRMellor**
**Documentation**

Updates the doc section on Managing TLS certificates.

- Removes [11.4. TLS connections](https://strimzi.io/docs/operators/latest/deploying.html#tls-connections-str)
- Consolidates content from TLS connections in the introduction to the section, using some of the content from the deleted section
- By removing the TLS Connections section, outdated references to TLS sidecars are also removed

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/assemblies/security/assembly-security.adoc [hunk 1]
```diff
@@ -8,28 +8,32 @@
...
 
-* Communication between Kafka and ZooKeeper
-* Interbroker communication between Kafka brokers
-* Internodal communication between ZooKeeper nodes
-* Strimzi operator communication with Kafka brokers and ZooKeeper nodes
+* Kafka brokers and ZooKeeper nodes
+* Kafka brokers (interbroker communication) 
+* ZooKeeper nodes (internodal communication)
+* Strimzi operators and Kafka and ZooKeeper
+* Cruise Control and Kafka and ZooKeeper
```
Comment (scholzj) [inline documentation/assemblies/security/assembly-security.adoc:17]:
```suggestion
* Cruise Control and Kafka
```

File: documentation/assemblies/security/assembly-security.adoc [hunk 2]
```diff
@@ -8,28 +8,32 @@
...
+* Kafka Exporter and Kafka and ZooKeeper
```
Comment (scholzj) [inline documentation/assemblies/security/assembly-security.adoc:18]:
```suggestion
* Kafka Exporter and Kafka
```

**Review by scholzj — APPROVED** [review]
Two nits. But LGTM otherwise.

**Review by ppatierno — APPROVED** [review]
LGTM. Apart Jakub's suggestions.

**Pull request merged**